### ISSUE TITLE
Temperature offset at pluto_temp.sh

### ISSUE THREAD

**Issue by Mane76** [analogdevicesinc/plutosdr_scripts#11]
When using pluto_temp.sh on a version "C" there is a temperature difference shown between FPGA and Tuner of approx 20°C. 
Even if the pluto is started from "cold" and massive heat-sinks are connected to FPGA and Tuner this temperature difference is shown - which seems not to be correct.

Any information to the background of this difference is highly welcome.

BR
Manfred

**Comment by mhennerich** [comment]
Hi Manfred,

First of all the temperature readings are die temperatures.
Even if you have a high thermal conductivity cooling solutions, the temperatures will never be the same.

Besides this the temp sensor on the AD936x TRX requires a one-point calibration, please see here: 
https://wiki.analog.com/resources/tools-software/linux-drivers/iio-transceiver/ad9361#die_temperature_reading

-Michael

**Comment by Mane76** [comment]
Hello Michael,
thank you for your fast reply - will work with that.

BR
Manfred

**Comment by Mane76** [comment]
Hello Michael, sorry to bother you again short before christmas:

I read the link about one point calibration and understood this is valid for the AD936x temp readings.
When I look in your code the lines related to the AD936x are

_pluto_temp=$(iio_attr ${uri} -c ad9361-phy temp0 input)
pluto=$(awk "BEGIN {printf(\"%.1f\", ${pluto_temp} / 1000)}")_

pluto is displayed as °C. Is the procedure described in the link done in the background ? So is the temp displayed already the "calibrated" temp reading ?

In contradiction, the lines in the code related to the FPGA / xadc show a work work with an offset

_xadc_raw=$(iio_attr ${uri} -c xadc temp0 raw)
xadc_offset=$(iio_attr ${uri} -c xadc temp0 offset)
xadc_scale=$(iio_attr ${uri} -c xadc temp0 scale)
xadc=$(awk "BEGIN {printf(\"%.1f\", (${xadc_raw} + ${xadc_offset}) * ${xadc_scale}/1000)}")_

xadc is dispayed as °C. 

So to sum up and in short words, do the both temps shown by the script represent the "correct" die temperatures will all "calibration" in it ?

BR
Manfred

**Comment by DG9BFC** [comment]
can i log into a pluto with putty and modify that offset?
if yes ... how?
BR sigi dg9bfc

**Comment by DG9BFC** [comment]
at best a step by step description how to "calibrate" and store in the pluto
BR dg9bfc sigi
